server_listener.go

```diff
@@ -0,0 +1,56 @@
+package golang
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+	"net"
+	"os"
+)
+
+// generateTLSConfig generates a new TLS config based on given
+// certificate path and key path.
+func generateTLSConfig(certPath, keyPath, caCertPath string) (*tls.Config, error) {
+	// Load certificate
+	certificate, err := tls.LoadX509KeyPair(certPath, keyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create certificate pool
+	certPool := x509.NewCertPool()
+	rootCert, err := ioutil.ReadFile(caCertPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Append cert to cert pool
+	ok := certPool.AppendCertsFromPEM(rootCert)
```

Comment:
This can be streamlined to `if ok := certPool.AppendCertsFromPEM(rootCert); !ok {...`